src/DurableTask.AzureStorage/Tracking/InstanceStoreBackedTrackingStore.cs

```
@@ -54,37 +56,33 @@ public override async Task<OrchestrationHistory> GetHistoryEventsAsync(string in
             }
 
             var events = await this.instanceStore.GetOrchestrationHistoryEventsAsync(instanceId, expectedExecutionId);
-
-            if (events == null || !events.Any())
-            {
-                return new OrchestrationHistory(EmptyHistoryEventList);
-            }
-            else
-            {
-                return new OrchestrationHistory(events.Select(x => x.HistoryEvent).ToList());
-            }
+            IList<HistoryEvent> history = events?.Select(x => x.HistoryEvent).ToList();
+            return new OrchestrationHistory(history ?? Array.Empty<HistoryEvent>());
         }
 
         /// <inheritdoc />
-        public override async Task<InstanceStatus> FetchInstanceStatusAsync(string instanceId)
+        public override async Task<InstanceStatus> FetchInstanceStatusAsync(string instanceId, CancellationToken cancellationToken = default)
         {
             OrchestrationState state = await this.GetStateAsync(instanceId, executionId: null);
             return state != null ? new InstanceStatus(state) : null;
         }
 
         /// <inheritdoc />
-        public override async Task<IList<OrchestrationState>> GetStateAsync(string instanceId, bool allExecutions, bool fetchInput = true)
+        public override async IAsyncEnumerable<OrchestrationState> GetStateAsync(string instanceId, bool allExecutions, bool fetchInput = true, [EnumeratorCancellation] CancellationToken cancellationToken = default)
```

I don't know if this is recommended or not, but I tend to _not_ expose the cancellation token publicly on async IAsyncEnumerable methods. Instead, I split into a public and private method:

``` CSharp
public IAsyncEnumerable<T> MethodName(...args)
{
    return MethodNameCore(args);
}

private async IAsyncEnumerable<T> MethodNameCore(...args, [EnumeratorCancellation] CancellationToken cancellationToken = default)
{
    // actual implementation
}
```

The reasoning is dotnet analyzers has a warning about always forwarding cancellation token. Which gets redundant with `System.Linq.Async` APIs that take in cancellation and also `IAsyncEnumerable.WithCancellation()` exists. Just a couple lines down that analyzer would flag and require you to write `await this.GetStateAsync(instanceId, false, cancellationToken).FirstOrDefaultAsync(cancellationToken);` A bit redundant right? By hiding the cancellation token from the public method, we get rid of the redundancy.

---

I think that makes perfect sense for conciseness. However, I am also seeing recent BCL additions that include an explicit token parameter, like in the `ChannelReader` type that was introduced with .NET 6 that defines the method [`ReadAllAsync`](https://github.com/dotnet/runtime/blob/d099f075e45d2aa6007a22b71b45a08758559f80/src/libraries/System.Threading.Channels/src/System/Threading/Channels/ChannelReader.netcoreapp.cs#L18). There is also [`DeserializeAsyncEnumerable`](https://github.com/dotnet/runtime/blob/5ed24429e72ea7ed7348a5b3698c82d1a13a602f/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Stream.cs#L366) in the `JsonSerializer` class. Although, there are not too many examples of `IAsyncEnumerable<T>` yet in the BCL.